### PR TITLE
Sysadmin deploy fixes

### PR DESCRIPTION
**Pull request by kotfic**
@aashish24 Could you please test and review.  Thanks!

### CODE REVIEW

File: ansible/Vagrantfile [hunk 1]
```diff
@@ -13,11 +13,12 @@ Vagrant.configure("2") do |config|
...
     geoservices.vm.provider :virtualbox do |vb|
        vb.customize [ "modifyvm", :id, "--name", "geoservices","--memory", 4096 ]
-  	end
+        end
     geoservices.vm.provision "ansible" do |ansible|
-        ansible.playbook = "playbook.yml"
-        ansible.verbose = "vvvv"
-        ansible.host_key_checking = false
+      ansible.playbook = "playbook.yml"
+      ansible.galaxy_role_file = "requirements.txt"
```
Comment (aashish24) [inline ansible/Vagrantfile:19]: Nice

File: ansible/ansible.cfg [hunk 2]
```diff
@@ -4,3 +4,4 @@ host_key_checking=False
 record_host_keys=False
 ssh.insert_key=False
 roles_path=roles
+ask_sudo_pass=True
```
Comment (aashish24) [inline ansible/ansible.cfg:7]: Nice, I am assuming that this is needed for local and for configuration against existing vms?

File: ansible/playbook.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 - hosts: geoservices
-  remote_user: vagrant
+  remote_user: "{{ deploy_user }}"
```
Comment (aashish24) [inline ansible/playbook.yml:2]: +1

File: ansible/roles/nginx/templates/nginx.conf.j2 [hunk 4]
```diff
@@ -4,6 +4,9 @@ server {
 
     charset     utf-8;
 
+    server_tokens off;
+    more_set_headers 'Server: Kitware ;)';
```
Comment (aashish24) [inline ansible/roles/nginx/templates/nginx.conf.j2:8]: What is this for @kotfic ?

File: ansible/playbook.yml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 - hosts: geoservices
```
Comment (aashish24) [inline ansible/playbook.yml:1]: Should be make hosts configurable as well? so instead of hosts: geoservices 

```
hosts: "{{ deploy_hosts }}"
```

**Comment by aashish24** [comment]
@kotfic looking good, just had one suggestion and now I am trying it locally. I will let you know how it goes.

**Comment by aashish24** [comment]
@kotfic I ran it locally and found one issue (please see below). Looks like geonode's user is still vagrant. 

TASK [geonode : checkout latest web app code] **********************************
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "/usr/bin/git clone --origin origin --branch 2.4.x https://github.com/GeoNode/geonode.git /home/vagrant/geonode", "failed": true, "msg": "fatal: could not create leading directories of '/home/vagrant/geonode': Permission denied", "rc": 128, "stderr": "fatal: could not create leading directories of '/home/vagrant/geonode': Permission denied\n", "stdout": "", "stdout_lines": []}

**Comment by aashish24** [comment]
@kotfic found some issues: 

**[chaudhary@einstein ansible (master)]$ vagrant up geoservices**
Bringing machine 'geoservices' up with 'virtualbox' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

ansible provisioner:
- The following settings shouldn't exist: galaxy_role_file

**Comment by aashish24** [comment]
When running locally, found this issue: 

TASK [geonode : create the db datastore in geoserver] **************************
task path: /home/chaudhary/tools/dataqs/dataqs/ansible/roles/geonode/tasks/main.yml:170
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["/home/chaudhary/venvs/geonode/bin/python", "/home/chaudhary/geonode/create_db_store.py"], "delta": "0:00:00.038794", "end": "2016-04-18 10:58:38.881392", "failed": true, "rc": 1, "start": "2016-04-18 10:58:38.842598", "stderr": "Traceback (most recent call last):\n  File \"/home/chaudhary/geonode/create_db_store.py\", line 4, in <module>\n    ds = cat.create_datastore('geonode','geonode')\n  File \"/usr/lib/python2.7/dist-packages/geoserver/catalog.py\", line 314, in create_datastore\n    workspace = self.get_workspace(workspace)\n  File \"/usr/lib/python2.7/dist-packages/geoserver/catalog.py\", line 865, in get_workspace\n    candidates = [w for w in self.get_workspaces() if w.name == name]\n  File \"/usr/lib/python2.7/dist-packages/geoserver/catalog.py\", line 861, in get_workspaces\n    description = self.get_xml(\"%s/workspaces.xml\" % self.service_url)\n  File \"/usr/lib/python2.7/dist-packages/geoserver/catalog.py\", line 216, in get_xml\n    raise FailedRequestError(\"Tried to make a GET request to %s but got a %d status code: \n%s\" % (rest_url, response.status, content))\ngeoserver.catalog.FailedRequestError: Tried to make a GET request to http://localhost:8080/geoserver/rest/workspaces.xml but got a 404 status code: \n<html><head><title>Apache Tomcat/7.0.52 (Ubuntu) - Error report</title><style></style> </head><body><h1>HTTP Status 404 - /geoserver/rest/workspaces.xml</h1><HR size=\"1\" noshade=\"noshade\"><p><b>type</b> Status report</p><p><b>message</b> <u>/geoserver/rest/workspaces.xml</u></p><p><b>description</b> <u>The requested resource is not available.</u></p><HR size=\"1\" noshade=\"noshade\"><h3>Apache Tomcat/7.0.52 (Ubuntu)</h3></body></html>", "stdout": "", "stdout_lines": [], "warnings": []}
...ignoring

File: ansible/roles/nginx/templates/nginx.conf.j2 (hunk 4 above)
Comment (kotfic) [inline ansible/roles/nginx/templates/nginx.conf.j2:8]: Sysadmin asked for this line to be added - i'm not sure what it is for

File: ansible/playbook.yml (hunk 5 above)
Comment (kotfic) [inline ansible/playbook.yml:1]: unfortunately this is not how ansible works - we can use the inventory to put whatever hosts we want into the geoservices group,  but the plays must have statically defined host groups.

**Comment by kotfic** [comment]
vagrant is the default for the `deploy_user` - it has to be set to a different value if the vagrant user isn't the correct deploy user.  this can be done on the command line by adding something like `-e deploy_user=ubuntu` if you want the deploy_user to be ubuntu

**Comment by kotfic** [comment]
from that error message there appears to be something wrong with the tomcat server - this is maybe related to the oracle java issue?

File: ansible/playbook.yml (hunk 5 above)
Comment (aashish24) [inline ansible/playbook.yml:1]: sure, makes sense, I forgot about it. Ignore my previous comment.

File: ansible/roles/nginx/templates/nginx.conf.j2 (hunk 4 above)
Comment (aashish24) [inline ansible/roles/nginx/templates/nginx.conf.j2:8]: It would be good to add some documentation here. Also would it affect if we deploy it on aws?

**Pull request merged**